### PR TITLE
Add SynthsUSD integration (L1 and L2) tests

### PR DESCRIPTION
**Pull request by leomassazza**
This PR adds integration tests for SynthsUSD (checks ERC20 behaviour) to integration tests for L1 and L2. It updates itBehavesLikeAnERC20 to support specifying a contract

### CODE REVIEW

File: test/integration/behaviors/erc20.behavior.js [hunk 1]
```diff
@@ -1,36 +1,50 @@
 const ethers = require('ethers');
 const { assert } = require('../../contracts/common');
+const { ensureBalance } = require('../utils/balances');
 
-function itBehavesLikeAnERC20({ ctx }) {
+function itBehavesLikeAnERC20({ ctx, contract }) {
 	describe('erc20 functionality', () => {
 		let owner, user;
-		let Synthetix;
+		let Contract;
```
Comment (ajsantander) [inline test/integration/behaviors/erc20.behavior.js:8]: Can we name this "token"?

File: test/integration/behaviors/erc20.behavior.js [hunk 2]
```diff
@@ -1,36 +1,50 @@
...
 		const amountToTransfer = ethers.utils.parseEther('1');
 
-		before('target contracts and users', () => {
-			({ Synthetix } = ctx.contracts);
+		before('target contracts and users', async () => {
+			if (contract) {
+				Contract = ctx.contracts[contract];
+				const symbol = await Contract.symbol();
+
+				await ensureBalance({
```
Comment (ajsantander) [inline test/integration/behaviors/erc20.behavior.js:19]: Can we put this in a separate before? I.e. "ensure owner has tokens"

File: test/integration/behaviors/erc20.behavior.js [hunk 3]
```diff
@@ -1,36 +1,50 @@
...
+				Contract = Contract.connect(owner);
 
-				const tx = await Synthetix.transfer(user.address, amountToTransfer);
+				const tx = await Contract.transfer(user.address, amountToTransfer);
 				await tx.wait();
 			});
 
-			it('increases the users balance', async () => {
-				assert.bnEqual(await Synthetix.balanceOf(user.address), userBalance.add(amountToTransfer));
+			it(`increases the users balance`, async () => {
```
Comment (ajsantander) [inline test/integration/behaviors/erc20.behavior.js:46]: Nit: no need for `, can use '

**Review by ajsantander — APPROVED** [review]
Awesome! Approved with requests for minor nit changes

File: test/integration/behaviors/erc20.behavior.js (hunk 1 above)
Comment (leomassazza) [reply test/integration/behaviors/erc20.behavior.js]: Done

File: test/integration/behaviors/erc20.behavior.js (hunk 2 above)
Comment (leomassazza) [reply test/integration/behaviors/erc20.behavior.js]: sure, it makes sense...

File: test/integration/behaviors/erc20.behavior.js (hunk 3 above)
Comment (leomassazza) [reply test/integration/behaviors/erc20.behavior.js]: my bad :(

File: test/integration/utils/users.js [hunk 4]
```diff
@@ -13,6 +13,7 @@ async function loadUsers({ ctx, network }) {
 	ctx.users = {};
 	ctx.users.owner = wallets[0];
 	ctx.users.someUser = wallets[1];
+	ctx.users.otherUser = wallets[2];
```
Comment (leomassazza) [inline test/integration/utils/users.js:16]: This is a change for an upcoming addition where we can need more than one user (transfer from-to)

**Pull request merged**